File: models/category.go
package models

// Category represents a channel category returned by the API.
type Category struct {
	ID        string `json:"id"`
	Name      string `json:"name"`
	Position  int    `json:"position"`
	CreatedAt string `json:"created_at"`
	UpdatedAt string `json:"updated_at"`
}

// CreateCategoryRequest is the body for POST /api/v1/server/categories.
type CreateCategoryRequest struct {
	Name string `json:"name"`
}

// UpdateCategoryRequest is the body for PATCH /api/v1/categories/:categoryID.
type UpdateCategoryRequest struct {
	Name     *string `json:"name"`
	Position *int    `json:"position"`
}

File: models/category.ts
/** Channel category returned by the API. */
export interface Category {
  id: string;
  name: string;
  position: number;
  created_at: string;
  updated_at: string;
}

/** Body for POST /api/v1/server/categories. */
export interface CreateCategoryRequest {
  name: string;
}

/** Body for PATCH /api/v1/categories/:categoryID. */
export interface UpdateCategoryRequest {
  name?: string | null;
  position?: number | null;
}
